src/csharp/Microsoft.Spark/ML/Linalg/SparseVectorUDT.cs

```diff
@@ -0,0 +1,75 @@
+using Microsoft.Spark.Sql.Types;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Spark.ML.Linalg
+{
+    /// <summary>
+    /// The data type for a SparseVectorUDT, used by SparseVector. 
+    /// </summary>
+    public class SparseVectorUDT : DataType
+    {
+        /// <summary>
+        /// Returns JSON object describing this type.
+        /// </summary>
+        internal override object JsonValue =>
+            new JObject(
+                new JProperty("type", "udt"),
+                new JProperty("class", "org.apache.spark.ml.linalg.VectorUDT"),
+                new JProperty("pyClass", "pyspark.ml.linalg.VectorUDT"),
+                new JProperty("sqlType", new JObject(
+                    new JProperty("type", "struct"),
+                    new JProperty("fields", new JArray(
+                            new JObject(
+                                new JProperty("name", "type"),
+                                new JProperty("type", "byte"),
+                                new JProperty("nullable", false)
+                                ),
+                            new JObject(
+                                new JProperty("name", "size"),
+                                new JProperty("type", "integer"),
+                                new JProperty("nullable", false)
+                            ),
+                            new JObject(
+                                new JProperty("name", "indices"),
+                                new JProperty("type", new JObject(
+                                    new JProperty("type", "array"),
+                                    new JProperty("elementType", "integer"),
+                                    new JProperty("containsNull", false))),
+                                new JProperty("nullable", true)
+                            ),
+                            new JObject(
+                                new JProperty("name", "values"),
+                                new JProperty("type", new JObject(
+                                    new JProperty("type", "array"),
+                                    new JProperty("elementType", "double"),
+                                    new JProperty("containsNull", false))),
+                                new JProperty("nullable", true)
+                            )
+                        ))
+                    ))
+                );
+
+        internal override object FromInternal(object obj)
+        {
+            if (obj is RowConstructor rowConstructor)
+            {
+                return rowConstructor.GetRow();
+            }
+
+            return obj;
+        }
+
+        internal override bool NeedConversion() => true;
+
+        /// <summary>
+        /// Returns a readable string that represents this type.
+        /// </summary>
+        public override string SimpleString =>
+            "struct<`type`:TINYINT,`size`:INT,`indices`:ARRAY<INT>,`values`:ARRAY<DOUBLE>>";
```

Comment:
shouldn't this be `vector` ?

Reply:
I don't *think* so (I could be wrong!) , in PySpark they have a VectorUDT which has to be used and this is the schema that VectorUDT returns (https://github.com/apache/spark/blob/master/mllib/src/main/scala/org/apache/spark/ml/linalg/VectorUDT.scala) see https://stackoverflow.com/questions/51527066/vectortype-for-structtype-in-pyspark-schema

Reply:
I'm a bit confused of how you're using this DataType.  Can you give some examples of its usage in the PR, maybe in the Tests ?

Also if we follow the pattern that's been used when creating `DataType`s, then shouldn't `SimpleString` should be the one defined [here] ?(https://github.com/apache/spark/blob/9db566a8821c02427434c551ee6e4d2501563dfa/python/pyspark/ml/linalg/__init__.py#L165)

Reply:
I've changed the simple string to match the value in pyspark now.

The data type is shown here in the test:

https://github.com/dotnet/spark/pull/810/files?file-filters%5B%5D=.cs#diff-129fac0a213898cb61e61adff29463de6deac7ab63ec58aa7784f02df20eecd9R41

The point of it is to stop users having to define the schema themselves - the classes are a bit of a pain to work with so having to define the schema for  Sparse/Dense Vector/Matrix is annoying - if you think that they are best left out of here then I can take it out again.

Reply:
Okay I see the example now.   Can you modify the usage so users won't have to explicitly call `SparseVector.FromRow` ?

Reply:
I’ll have another look at it, it was a while ago when I tried and there was something stopping it but I can’t exactly remember what!